lib/services/chat_bot_service.dart

```diff
@@ -0,0 +1,114 @@
+import 'dart:collection';
+import 'package:summer2022/exceptions/invalid_command_exception.dart';
+import 'package:summer2022/models/ApplicationFunction.dart';
+import 'package:summer2022/services/bases/chat_bot.dart';
+import 'package:summer2022/utility/RouteGenerator.dart';
+
+class ChatBotService implements ChatBot {
+  Map<SiteAreas, List<String>> ChatFunctions = HashMap();
+
+  /**
+   * Default constructor
+   */
+  ChatBotService() {
+    _populateChatFunctions();
+  }
+
+  /**
+   * Perform user entered chat function
+   * returns an application function to be called by UI
+   */
+  @override
+  ApplicationFunction performChatFunction(SiteAreas currentArea, String userInput) {
+    var parsedInput = userInput.split(' ');
+
+    try {
+      // Attempt to retrieve the first word (command)
+      var command = ChatFunctions[currentArea]!.toList();
```

Comment:
Is `ChatFunctions[currentArea]` guaranteed to be non-null?
If so, this is fine, otherwise we should probably check against null and use an empty list or something.

Reply:
Will check if key is valid on entry into the method  and return message stating there are no commands